compiler-rt/test/builtins/Unit/extendhfxf2_test.c

```diff
@@ -5,59 +5,55 @@
 #include <math.h> // for isnan, isinf
 #include <stdio.h>
 
-#include "int_lib.h"
+#include "fp_test.h"
 
 #if HAS_80_BIT_LONG_DOUBLE && defined(COMPILER_RT_HAS_FLOAT16)
 
-long double __extendhfxf2(_Float16 f);
+xf_float __extendhfxf2(TYPE_FP16 f);
 
-int test_extendhfxf2(_Float16 a, long double expected) {
-  long double x = __extendhfxf2(a);
-  __uint16_t *b = (void *)&a;
-  int ret = !((isnan(x) && isnan(expected)) || x == expected);
+int test_extendhfxf2(TYPE_FP16 a, xf_float expected) {
+  xf_float x = __extendhfxf2(a);
+  uint16_t a_rep = toRep16(a);
+  int ret = compareResultF80(x, expected);
```

Comment:
this expects hi and lo. I'll have to change this.

Reply:
f80 functions in fp_test are only for x86_64

Reply:
Ah yes that should be relaxed to just be gated on fp80 support

Reply:
I.e. change the if in fp_test.h to just HAS_80_BIT_LONG_DOUBLE

Reply:
I made the test exclusive for x86_64. Would `__uint128_t` be available on i386?

Reply:
It may or may not - but we shouldn't need it for this. I'll take care of that as a follo-up.